src/wolfio.c

```diff
@@ -141,34 +141,35 @@ int BioReceive(WOLFSSL* ssl, char* buf, int sz, void* ctx)
     recvd = wolfSSL_BIO_read(ssl->biord, buf, sz);
     if (recvd <= 0) {
         if (ssl->biord->type == WOLFSSL_BIO_SOCKET) {
-            int err;
-
             if (recvd == 0) {
                 WOLFSSL_MSG("BioReceive connection closed");
                 return WOLFSSL_CBIO_ERR_CONN_CLOSE;
             }
-
-            err = wolfSSL_LastError(recvd);
-            if (err == SOCKET_EWOULDBLOCK || err == SOCKET_EAGAIN) {
-                WOLFSSL_MSG("\tWould block");
-                return WOLFSSL_CBIO_ERR_WANT_READ;
-            }
-            else if (err == SOCKET_ECONNRESET) {
-                WOLFSSL_MSG("\tConnection reset");
-                return WOLFSSL_CBIO_ERR_CONN_RST;
-            }
-            else if (err == SOCKET_EINTR) {
-                WOLFSSL_MSG("\tSocket interrupted");
-                return WOLFSSL_CBIO_ERR_ISR;
-            }
-            else if (err == SOCKET_ECONNABORTED) {
-                WOLFSSL_MSG("\tConnection aborted");
-                return WOLFSSL_CBIO_ERR_CONN_CLOSE;
-            }
+    #ifdef USE_WOLFSSL_IO
```

Comment:
Tree in forest scenario. If `#ifndef NO_BIO` is above and `NO_BIO` without `USE_WOLFSSL_IO` errors out on compilation then do we need this check here?